### PR TITLE
APP-75 - Longevity integration: unit tests

### PR DESCRIPTION
**Pull request by asurkov**

### CODE REVIEW

File: src/test/kotlin/app/CodeLongevityTest.kt [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        assertRevCommitLine(fromCommit, fromFile, fromLineNum, actualLine.from,
+                            "'$lineText' from_commit");
+        assertRevCommitLine(toCommit, toFile, toLineNum, actualLine.to,
+                            "'$lineText' to_commit");
+        assertEquals(lineText, actualLine.text, "line text")
+    }
+
+    given("'test group #1'") {
+        val testRepoPath = "../testrepo1"
+        val testRepo = TestRepo("../testrepo1")
```
Comment (anatolykislov) [inline src/test/kotlin/app/CodeLongevityTest.kt]: TestRepo(testRepoPath)

File: src/test/kotlin/app/CodeLongevityTest.kt [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+                           lines2[2])
+        }
+
+        // t3: subsequent deletion
+        testRepo.deleteLines(fileName, 2, 2)
+        val rev3 = testRepo.commit("delete line")
+        val lines3 = CodeLongevity(LocalRepo(testRepoPath), Repo(),
+                                   MockApi(), testRepo.git).compute()
+
+        it("'t3: subsequent deletion'") {
```
Comment (anatolykislov) [inline src/test/kotlin/app/CodeLongevityTest.kt]: Quotes?

File: src/test/kotlin/app/CodeLongevityTest.kt [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            assertCodeLine("line in the middle",
+                           rev2, fileName, 1,
+                           rev4, fileName, 1,
+                           lines4[0])
+            assertCodeLine("line1",
+                           rev1, fileName, 0,
+                           rev4, fileName, 0,
+                           lines4[1])
+
+            /* TODO(alex) the test fails
```
Comment (anatolykislov) [inline src/test/kotlin/app/CodeLongevityTest.kt]: TODO(alex): the test fails.

File: src/test/kotlin/app/CodeLongevityTest.kt [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+import org.eclipse.jgit.revwalk.RevCommit
+
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.given
+import org.jetbrains.spek.api.dsl.it
+
+/**
+ * A wrapper around Git repo allowing to add/remove/edit files and make commits.
+ */
+class TestRepo(val repoPath: String) {
```
[the remark this answers is not in the document]
Comment (asurkov) [reply src/test/kotlin/app/CodeLongevityTest.kt]: Yeah, wasn't sure about host folder and file name. I was thinking about same folder and TestRepo.kt as a file name but I find it confusing to have TestRepo.kt for utils and TestNameTest.kt for tests. If we had something like test_TestName.kt for tests names, then it'd be nicer. Suggestions?

File: src/test/kotlin/app/CodeLongevityTest.kt (hunk 2 above)
Comment (asurkov) [reply src/test/kotlin/app/CodeLongevityTest.kt]: you mean single quotes? it just looks nicer when errors are reported in the console

Comment (anatolykislov) [reply src/test/kotlin/app/CodeLongevityTest.kt]: In first tests no single quotes.

**Pull request merged**